src/components/Products.js

```diff
@@ -27,6 +26,39 @@ function Proudects() {
       .catch((err) => setProducts([]));
   }, [setProducts]);
 
+  if (products) {
+    for (let i = 0; i <= 19; i++) {
+      products[i]["quantity"] = 1;
+    }
+    for (let i = 0; i <= 19; i++) {
+      products[i]["color"] = "black";
+    }
+  }
+
```

Comment:
The logic for setting a quantity for each product should be right after the products were fetched from the API (in the get request useEffect).

In addition, the for loop should loop till product.length not 19